controllers/openstackcluster_controller.go

```diff
@@ -221,6 +221,16 @@ func deleteBastion(scope scope.Scope, cluster *clusterv1.Cluster, openStackClust
 		return err
 	}
 
+	if openStackCluster.Status.Bastion != nil {
+		for _, port := range openStackCluster.Status.Bastion.PortsStatus {
+			if err := networkingService.DeleteInstanceTrunkAndPort(openStackCluster, port); err != nil {
+				handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete port: %w", err))
+				return fmt.Errorf("failed to delete port: %w", err)
+			}
+		}
+		openStackCluster.Status.Bastion.PortsStatus = nil
+	}
```

Comment:
I've tried to delete the VM first and then the ports and e2e job was green. Maybe it's ok to do this? Since we create port firsts then instances, we should probably remove instances first (where Nova will detach the ports anyway) and then remove the ports. WDYT?

Reply:
I don't see any problem with that.